node/src/daemon/setup_reporter.rs

```diff
@@ -801,16 +816,21 @@ impl ValueRetriever for MappingProtocol {
 
     fn computed_default(
         &self,
-        _bootstrapper_config: &BootstrapperConfig,
+        bootstrapper_config: &BootstrapperConfig,
         persistent_config: &dyn PersistentConfiguration,
         _db_password_opt: &Option<String>,
     ) -> Option<(String, UiSetupResponseValueStatus)> {
-        let persistent_config_value_opt = match persistent_config.mapping_protocol() {
-            Ok(protocol_opt) => protocol_opt,
-            Err(_) => None,
-        };
-        persistent_config_value_opt
-            .map(|protocol| (protocol.to_string().to_lowercase(), Configured))
+        match (
+            bootstrapper_config.mapping_protocol_opt,
+            persistent_config.mapping_protocol(),
+        ) {
+            (_, Err(e)) => todo!("Error retrieving mapping protocol from database: {:?}", e),
+            (Some(from_config), _) => Some((from_config.to_string().to_lowercase(), Configured)),
```

Comment:
This always make me wonder. This is definitely one of the least clear parts in `setup_reporter`. Correct me if my interpretations differs from yours and you think you're right. 

Because we are dealing with `computed_default()` it's just another layer of setup clusters which will eventually be combined together according to their respective priorities. I believe that if this value is set on the command line it has nothing to do with what we're doing here.

Instead, this place is supposed to check only the remaining potential sources as the config file, environment variables and last but not least the database. All these then are marked with Configured if detected being populated. 

That's the reason why we say `Configured` but not `Set` on this line above. Right?

Reply:
Yes. `computed_default()` should never mark anything `Set`: only `Default` or `Configured`.